src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/ResendStaffNotificationsCoreCaseDataTest.java

```diff
@@ -0,0 +1,248 @@
+        given(userService.getUserDetails(AUTHORISATION_TOKEN)).willReturn(userDetails);
+        given(userService.authenticateAnonymousCaseWorker()).willReturn(user);
+        given(userService.getUser(AUTHORISATION_TOKEN)).willReturn(user);
+        given(authTokenGenerator.generate()).willReturn(SERVICE_TOKEN);
+    }
+
+    @Test
+    public void shouldRespond404WhenClaimDoesNotExist() throws Exception {
+        final String nonExistingClaimReference = "something";
+        final String event = "claim-issue";
+        given(coreCaseDataApi.searchForCitizen(
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            eq(ImmutableMap.of("case.referenceNumber", CASE_REFERENCE, "page", PAGE, "sortDirection", "desc"))
+            )
+        ).willReturn(listOfCaseDetails());
+        makeRequest(nonExistingClaimReference, event).andExpect(status().isNotFound());
+        verify(emailService, never()).sendEmail(any(), any());
+    }
+
+    @Test
+    public void shouldRespond404WhenEventIsNotSupported() throws Exception {
+        final String nonExistingEvent = "some-event";
+
+        given(coreCaseDataApi.searchForCitizen(
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            eq(ImmutableMap.of("case.referenceNumber", CASE_REFERENCE, "page", PAGE, "sortDirection", "desc"))
+            )
+        ).willReturn(listOfCaseDetails());
+        makeRequest(CASE_REFERENCE, nonExistingEvent).andExpect(status().isNotFound());
+        verify(emailService, never()).sendEmail(any(), any());
+    }
+
+    @Test
+    public void shouldRespond409AndNotProceedForClaimIssuedEventWhenClaimIsLinkedToDefendant() throws Exception {
+        final String event = "claim-issued";
+
+        given(coreCaseDataApi.searchForCitizen(
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            eq(ImmutableMap.of("case.referenceNumber", CASE_REFERENCE, "page", PAGE, "sortDirection", "desc"))
+            )
+        ).willReturn(listOfCaseDetailsWithDefendant());
+
+        makeRequest(CASE_REFERENCE, event).andExpect(status().isConflict());
+
+        verify(emailService, never()).sendEmail(any(), any());
+    }
+
+    @Test
+    public void shouldRespond200AndSendNotificationsForClaimIssuedEvent() throws Exception {
+        final String event = "claim-issued";
+        final Claim claim = SampleClaim.builder()
+            .withReferenceNumber(CASE_REFERENCE)
+            .withDefendantId(null).build();
+
+        GeneratePinResponse pinResponse = new GeneratePinResponse("pin-123", "333");
+        given(userService.generatePin(anyString(), eq(AUTHORISATION_TOKEN))).willReturn(pinResponse);
+        given(sendLetterApi.sendLetter(any(), any())).willReturn(new SendLetterResponse(UUID.randomUUID()));
+        given(coreCaseDataApi.searchForCitizen(
```

Comment:
I would appreciate if you moved it before the mocks for PIN generation to have mocks in the same order as they happen in real application.